webapp/app/.server/service/signatures.ts

```diff
@@ -67,6 +61,87 @@ async function verifySignature({
       [action]: types,
     },
   });
+}
+
+async function isValidSignatureZkFoundation(
+  foundationAddress: Hex,
+  signature: Hex,
+  verifierAddr: Hex,
+  action: SignAction,
+  message: Record<string, string>,
+  types: { name: string; type: string }[],
+  contractName: string
+): Promise<boolean> {
+  const digest = createTypedDigest({
+    verifierAddr,
+    action,
+    message,
+    types,
+    contractName,
+  });
+  const code = await l1Rpc.getByteCode(foundationAddress);
+
+  if (code === undefined) {
+    return verifyTypedData({
+      domain: {
+        name: contractName,
+        version: "1",
+        chainId: env.ETH_NETWORK === "mainnet" ? mainnet.id : sepolia.id,
+        verifyingContract: verifierAddr,
+      },
+      primaryType: action,
+      message,
+      types: {
+        [action]: types,
+      },
+      signature,
+      address: foundationAddress,
+    });
+  }
+
+  const IERC1271Abi: AbiFunction = {
+    name: "isValidSignature",
+    inputs: [
+      { name: "hash", type: "bytes32" },
+      { name: "signature", type: "bytes" },
+    ],
+    outputs: [{ name: "magicValue", type: "bytes4" }],
+    type: "function",
+    stateMutability: "view",
+  };
```

Comment:
I'm just flagging this to be included in the new `@repo/contracts` package upcoming in the next version.

Reply:
We agreed on leaving this for next release.